lib/excon/socket.rb

```diff
@@ -229,6 +149,112 @@ def connect
       end
     end
 
+    def read_nonblock(max_length)
+      begin
+            if max_length
+              until @read_buffer.length >= max_length
+                @read_buffer << @socket.read_nonblock(max_length - @read_buffer.length)
+              end
+            else
+              loop do
+                @read_buffer << @socket.read_nonblock(@data[:chunk_size])
+              end
+            end
+          rescue OpenSSL::SSL::SSLError => error
+            if error.message == 'read would block'
+              timeout_reached('read') ? raise_timeout_error('read') : retry
```

Comment:
One, hopefully final thing. I think I'd prefer for these ternaries to be expanded also. Otherwise it's looking spot on. Thanks!